component/client/ClientChat.jsx

```diff
@@ -1,14 +1,32 @@
 import React, {Component} from "react";
 import {Panel} from "react-bootstrap";
 import ClientMessenger from "./ClientMessenger.jsx";
+import RoomService from "./RoomService";
 
 class ClientChat extends Component {
+
+  room: RoomService;
+
   constructor(props) {
     super(props);
+    this.state = {
+      history: []
+    };
+    // set up Room Service
+    this.room = new RoomService(props.url, (message) => {
```

Comment:
You might also want to add boilerplate for lifecycle method `componentWillReceiveProps(nextProps)`. This may help hotswap the webpage url so that the client is connected to a different chat room.